spec/forms/applicants/basic_details_form_spec.rb

```diff
@@ -266,6 +283,89 @@
     end
   end
 
+  context "with changed last name" do
+    let(:params) do
+      {
+        first_name: "Fred",
+        last_name: "Bloggs",
+        date_of_birth_1i: "1999",
+        date_of_birth_2i: "12",
+        date_of_birth_3i: "31",
+        changed_last_name: "true",
+        last_name_at_birth: "Smith",
+      }
+    end
+
+    it "is valid" do
+      expect(form).to be_valid
+    end
+
+    it "saves to the database" do
+      expect { form.save }.to change(Applicant, :count)
+    end
+  end
+
+  context "with changed_last_name nil" do
+    let(:params) do
+      {
+        first_name: "Fred",
+        last_name: "Bloggs",
+        date_of_birth_1i: "1999",
+        date_of_birth_2i: "12",
+        date_of_birth_3i: "31",
+        changed_last_name: nil,
+        last_name_at_birth: "",
+      }
+    end
+
+    it "adds expected error" do
+      form.save!
+      expect(form.errors[:changed_last_name]).to contain_exactly("Select yes if your client has ever changed their last name")
+    end
+  end
+
+  context "with changed_last_name but no last_name_at_birth" do
+    let(:params) do
+      {
+        first_name: "Fred",
+        last_name: "Bloggs",
+        date_of_birth_1i: "1999",
+        date_of_birth_2i: "12",
+        date_of_birth_3i: "31",
+        changed_last_name: true,
```

Comment:
This is a boolean but the other examples are a string. Without testing, I can't remember if it accepts both. Does it? Probably not an issue but flagging just in case

Reply:
I think the params are always strings, but it was passing because the form converts it to a string anyway before it checks it - have amended it anyway to be consistent with the other tests.